src/dmd/parse.d

```diff
@@ -448,21 +407,85 @@ class Parser(AST) : Lexer
         if (token.value == TOK.rightCurly)
         {
             error(token.loc, "unmatched closing brace");
-            goto Lerr;
+            return errorReturn();
         }
 
         if (token.value != TOK.endOfFile)
         {
             error(token.loc, "unrecognized declaration");
-            goto Lerr;
+            return errorReturn();
         }
         return decldefs;
+    }
 
-    Lerr:
-        while (token.value != TOK.semicolon && token.value != TOK.endOfFile)
-            nextToken();
-        nextToken();
-        return new AST.Dsymbols();
+
+    /**********************************
+     * Parse the ModuleAttributes preceding a module declaration.
+     * ModuleDeclaration:
+     *    ModuleAttributes(opt) module ModuleFullyQualifiedName ;
+     * https://dlang.org/spec/module.html#ModuleAttributes
+     * Params:
+     *  msg = set to the AssignExpression from DeprecatedAttribute https://dlang.org/spec/module.html#DeprecatedAttribute
+     *  isdeprecated = set to true if a DeprecatedAttribute is seen
+     */
+    private
```

Comment:
In the majority of cases, the dmd style is to put `private` on the same line with the function declaration.

Reply:
I've found this works better, as attributes can be added/removed without changing the signature. It also helps keep the line from getting too long.